### PR TITLE
[trendline] added vp9 10bit encode support

### PR DESCRIPTION
**Pull request by FocusLuo**
1, profile map
2, move [ffmpeg/qsv: move VP9-10 Encoder Test classes to lib]
3, add vp9-10 trendline modes

### CODE REVIEW

**Comment by uartie** [comment]
please fix conflicts and test failures

**Comment by FocusLuo** [comment]
> please fix conflicts and test failures

fixed and tested

File: lib/ffmpeg/qsv/util.py [hunk 1]
```diff
@@ -73,6 +73,10 @@ def mapprofile(codec, profile):
       "profile0"  : "profile0",
       "profile1"  : "profile1",
     },
+    "vp9-10" : {
+      "profile1"  : "profile1",
+      "profile2"  : "profile2",
+    },
```
Comment (uartie) [inline lib/ffmpeg/qsv/util.py:79]: profile1?  Should be profile2 and profile3, right?

File: test/ffmpeg-vaapi/encode/10bit/vp9.py [hunk 2]
```diff
@@ -16,7 +16,7 @@ class VP9_10EncoderBaseTest(EncoderTest):
   def before(self):
     super().before()
     vars(self).update(
-      codec    = "vp9",
+      codec    = "vp9-10",
```
Comment (uartie) [inline test/ffmpeg-vaapi/encode/10bit/vp9.py:19]: The lib/ffmpeg/vaapi/encoder.py has some conditional checks for self.codec "vp9".  Need to update them to include "vp9-10" with this change.

Also, mapprofile in ffmpeg/vaapi/util.py...

File: test/gst-msdk/encode/10bit/vp9.py [hunk 3]
```diff
@@ -16,7 +16,7 @@ class VP9_10EncoderBaseTest(EncoderTest):
   def before(self):
     super().before()
     vars(self).update(
-      codec         = "vp9",
+      codec         = "vp9-10",
```
Comment (uartie) [inline test/gst-msdk/encode/10bit/vp9.py:19]: The lib/gstreamer/msdk/encoder.py has some conditional checks for self.codec "vp9".  Need to update them to include "vp9-10" with this change.

File: test/gst-va/encode/10bit/vp9.py [hunk 4]
```diff
@@ -15,7 +15,7 @@ class VP9_10EncoderBaseTest(EncoderTest):
   def before(self):
     super().before()
     vars(self).update(
-      codec         = "vp9",
+      codec         = "vp9-10",
```
Comment (uartie) [inline test/gst-va/encode/10bit/vp9.py:18]: The lib/gstreamer/va/encoder.py has some conditional checks for self.codec "vp9".  Need to update them to include "vp9-10" with this change.

File: test/gst-vaapi/encode/10bit/vp9.py [hunk 5]
```diff
@@ -16,7 +16,7 @@ class VP9_10EncoderBaseTest(EncoderTest):
   def before(self):
     super().before()
     vars(self).update(
-      codec         = "vp9",
+      codec         = "vp9-10",
```
Comment (uartie) [inline test/gst-vaapi/encode/10bit/vp9.py:19]: The lib/gstreamer/vaapi/encoder.py has some conditional checks for self.codec "vp9".  Need to update them to include "vp9-10" with this change.

**Review by uartie — CHANGES_REQUESTED** [review]

File: lib/ffmpeg/qsv/util.py (hunk 1 above)
Comment (FocusLuo) [reply lib/ffmpeg/qsv/util.py]: You are right, updated

File: test/gst-msdk/encode/10bit/vp9.py (hunk 3 above)
Comment (FocusLuo) [reply test/gst-msdk/encode/10bit/vp9.py:19]: updated

File: test/gst-vaapi/encode/10bit/vp9.py (hunk 5 above)
Comment (FocusLuo) [reply test/gst-vaapi/encode/10bit/vp9.py:19]: updated

File: test/gst-va/encode/10bit/vp9.py (hunk 4 above)
Comment (FocusLuo) [reply test/gst-va/encode/10bit/vp9.py:18]: updated, thanks

File: test/ffmpeg-vaapi/encode/10bit/vp9.py (hunk 2 above)
Comment (FocusLuo) [reply test/ffmpeg-vaapi/encode/10bit/vp9.py:19]: updated, thanks

**Comment by uartie** [comment]
Thanks for update.  Please fix conflicts

**Comment by FocusLuo** [comment]
> Thanks for update. Please fix conflicts

fixed conflicts

**Pull request merged**